[Flang][OpenMP] Add lowering support for is_device_ptr clause

TIFitis opened the pull request:
Add support for OpenMP is_device_ptr clause for target directives.

TIFitis committed:
[Flang][OpenMP] Add lowering support for is_device_ptr clause

Add support for OpenMP is_device_ptr clause for target directives.

ergawy commented on flang/lib/Lower/OpenMP/OpenMP.cpp:
```diff
@@ -2531,6 +2533,41 @@ genTargetOp(lower::AbstractConverter &converter, lower::SymMap &symTable,
+      if (!sym || !mapVal)
+        continue;
+      auto mapInfo = mapVal.getDefiningOp<mlir::omp::MapInfoOp>();
+      if (!mapInfo)
```
Same question as above.

TIFitis replied:
Fixed

ergawy reviewed (commented):
Thanks Akash! Just a few stop-by comments.

TIFitis committed:
Address reviewer comments.

TIFitis commented on flang/test/Lower/OpenMP/target.f90:
```diff
@@ -566,6 +566,36 @@ subroutine omp_target_device_addr
+
+!CHECK-LABEL: func.func @_QPomp_target_is_device_ptr() {
+subroutine omp_target_is_device_ptr
+   use iso_c_binding, only: c_associated, c_ptr
```
Done

ergawy reviewed (approved):
LGTM, thanks Akash. Please wait for another approval before merging.

TIFitis committed:
[MLIR][OpenMP] Add OpenMPToLLVMIRTranslation support for is_device_ptr (#169367)

This PR adds support for the OpenMP `is_device_ptr` clause in the MLIR
to LLVM IR translation for target regions. The `is_device_ptr` clause
allows device pointers (allocated via OpenMP runtime APIs) to be used
directly in target regions without implicit mapping.

TIFitis committed:
Address reviewer comment.

TIFitis committed:
Merge branch 'main' into users/Akash/is_device_ptr_lower

mjklemm reviewed (approved):
LGTM
